pkg/kv/kvserver/replica_consistency.go

```diff
@@ -581,6 +594,11 @@ func CalcReplicaDigest(
 	if !statsOnly {
 		ms, err := rditer.ComputeStatsForRangeWithVisitors(&desc, snap, 0, /* nowNanos */
 			pointKeyVisitor, rangeKeyVisitor)
+		// Consume the remaining quota borrowed in the visitors. Do it even on
+		// iteration error, but prioritize returning the latter if it occurs.
+		if wErr := limiter.WaitN(ctx, batchSize); wErr != nil && err == nil {
```

Comment:
Can you verify that WaitN(ctx, 0) is valid? I'd expect it to, but worth confirming.

Reply:
I checked it yesterday, it returns immediately, which is what we need.